Develop

alex-from-90 opened the pull request:
add all branch

alex-from-90 committed:
ТЗ 12

nik110996 committed:
19.06.2023_First commit

alex-from-90 committed:
ТЗ 12.01 refactor add Rewrite controller and Service

Rat-In-Hat committed:
добавить ветку в реп

nik110996 committed:
19.06.2023_Second commit

nik110996 committed:
19.06.2023_Second commit2

nik110996 committed:
19.06.2023_Second commit3

nik110996 committed:
19.06.2023_Second commit3

nik110996 committed:
Merge remote-tracking branch 'origin/add-recommendations' into add-recommendations

nik110996 committed:
Merge remote-tracking branch 'origin/add-recommendations' into add-recommendations

alex-from-90 committed:
ТЗ 12.02 works

nik110996 committed:
Merge remote-tracking branch 'origin/add-recommendations' into add-recommendations

nik110996 committed:
Merge remote-tracking branch 'origin/add-recommendations' into add-recommendations

EvgeniyBushuev committed:
обавлена сущность Feed (событие)

Rat-In-Hat committed:
del

удалить инмемори реализацю. функция удалить не возвращает значение

EvgeniyBushuev committed:
добавить эндпоинт на получение списка событий у пользователя

Rat-In-Hat committed:
style

применить код стайл к проекту

Rat-In-Hat committed:
ref, feat
создать пакет имплементации, перенести туда классы имплементирующие интерфейсы
добавить удаление юзера по id

Rat-In-Hat committed:
ref

поменять названия и расположение методов

Rat-In-Hat committed:
style

убрать лишний отступ

Rat-In-Hat committed:
fix

добавить анотацию

Rat-In-Hat committed:
style

убрать лишний отступ

nik110996 committed:
recommends by one request

nik110996 committed:
Merge remote-tracking branch 'origin/add-recommendations' into add-recommendations

nik110996 committed:
Merge remote-tracking branch 'origin/add-recommendations' into add-recommendations

EvgeniyBushuev committed:
реализовать функционал записи событий

nik110996 committed:
Merge remote-tracking branch 'origin/add-recommendations' into add-recommendations

EvgeniyBushuev committed:
добавить логирование к эндпоинту GET(/id/feed)

nik110996 committed:
Merge pull request #15 from alex-from-90/add-recommendations

19.06.2023_First commit

alex-from-90 committed:
resolve conflict

EvgeniyBushuev committed:
Мерж c веткой  develop

EvgeniyBushuev committed:
Без изменений

krakaduum committed:
Реализована функциональность «Фильмы по режиссёрам»

alex-from-90 committed:
ТЗ 12.02 add reviews

alex-from-90 committed:
ТЗ 12.02 Code Style

Rat-In-Hat committed:
Merge branch 'develop' into add-remove-endpoint

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/service/UserService.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/inMemory/InMemoryFilmStorage.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/interfaces/FilmStorage.java
#	src/test/java/ru/yandex/practicum/filmorate/db/FilmorateApplicationTests.java

Rat-In-Hat committed:
fix

развернуть пакет

Rat-In-Hat committed:
fix

вернуть инмемори.

Rat-In-Hat committed:
style

применен код стайл

Rat-In-Hat committed:
fix
исправить название метода в тестах

alex-from-90 committed:
ТЗ 12.03 remove mapper, add merge, use BeanPropertyRowMapper

alex-from-90 committed:
ТЗ 12.04 remove like after dislike

DeepDarkFanta committed:
save

DeepDarkFanta committed:
Merge branch 'develop' into add-search

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/storage/FilmDbStorage.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/inMemory/InMemoryFilmStorage.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/interfaces/FilmStorage.java
#	src/main/resources/schema.sql

alex-from-90 committed:
ТЗ 12.05 refactoring

EvgeniyBushuev committed:
Merge pull request #16 from alex-from-90/add-remove-endpoint

add-remove-endpoint

EvgeniyBushuev committed:
Merge branch 'develop' into add-feed

alex-from-90 committed:
Merge branch 'develop' into add-reviews

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java

alex-from-90 committed:
ТЗ 12.06 add from develop

EvgeniyBushuev committed:
Решить конфликты после мержа

alex-from-90 committed:
Merge pull request #17 from alex-from-90/add-reviews

Add reviews

EvgeniyBushuev committed:
Merge branch 'develop' into add-feed

krakaduum committed:
Merge branch 'develop' into add-director

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java
#	src/main/resources/schema.sql

DeepDarkFanta committed:
Merge pull request #18 from alex-from-90/add-director

Реализована функциональность «Фильмы по режиссёрам»

DeepDarkFanta committed:
save

DeepDarkFanta committed:
Merge branch 'develop' into add-search

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/inMemory/InMemoryFilmStorage.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/interfaces/FilmStorage.java
#	src/main/resources/schema.sql

DeepDarkFanta committed:
add search by title,director

DeepDarkFanta committed:
add search

EvgeniyBushuev committed:
Тестирование событий в ReviewService

EvgeniyBushuev committed:
Тестирование событий в ReviewService

EvgeniyBushuev committed:
Тестирование событий в ReviewService

EvgeniyBushuev committed:
Победа над тестами событий для отзывов

EvgeniyBushuev committed:
Merge branch 'develop' into add-feed

EvgeniyBushuev committed:
Merge pull request #20 from alex-from-90/add-search

Add search

EvgeniyBushuev committed:
Merge branch 'develop' into add-feed

EvgeniyBushuev committed:
fix: CodeStyle

EvgeniyBushuev committed:
fix: CodeStyle

DayZscripter committed:
добавлен вывод топ-n фильмов по колличеству лайков

DayZscripter committed:
changes

EvgeniyBushuev committed:
fix: по комментариям PR

EvgeniyBushuev committed:
fix: по комментариям PR

DeepDarkFanta committed:
Merge pull request #24 from alex-from-90/add-feed

Add feed

DayZscripter committed:
Merge pull request #27 from alex-from-90/add-most-populars

Add most populars

DayZscripter committed:
крайние правки

DayZscripter committed:
мелкие правки

DayZscripter committed:
fix: LikeStorage sql request

DayZscripter committed:
fix: Update LikeStorage

DayZscripter committed:
Merge pull request #30 from alex-from-90/add-common-films

add-common-films

alex-from-90 committed:
compare and merge

alex-from-90 committed:
ТЗ 12.07 assembling

alex-from-90 committed:
ТЗ 12.07 assembling

alex-from-90 committed:
ТЗ 12.07 assembling and CodeStyle

alex-from-90 committed:
ТЗ 12.08 CodeStyle

EvgeniyBushuev committed:
fix: исправить метод getPopular

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -53,13 +56,35 @@ public void addLike(@PathVariable Long id, @PathVariable Long userId) {
-        log.info("Получен DELETE-запрос к эндпоинту: '/films' на удаление фильма с ID={}", id);
-        return filmService.delete(id);
+    @GetMapping("/director/{directorId}")
+    public List<Film> getDirectorFilms(@PathVariable int directorId, @RequestParam String sortBy) {
```
Тип параметра `sortBy `лучше переделать на enum, тогда валидация будет происходить на уровне контроллера

ImING0 replied:
Дарина реализовала енам

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -53,13 +56,35 @@ public void addLike(@PathVariable Long id, @PathVariable Long userId) {
+
+    @GetMapping("/search")
+    public List<Film> filmsSearch(@RequestParam String query,
+                                  @RequestParam(defaultValue = "") String by) {
```
Аналогично с параметром `by `- лучше enum

ImING0 replied:
Внедрил енам. Логику запроса и метода не затронул, просто заменил входные параметры

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/service/FeedService.java:
```diff
@@ -0,0 +1,30 @@
+
+        Feed feed = feedStorage.createFeed(userId, entityId, eventType, operation);
+
+        feedStorage.addFeed(feed);
```
вызов `feedStorage.addFeed()` кажется излишним, этот метод можно сделать приватным и вызывать в `createFeed()`

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/storage/FilmDbStorage.java:
```diff
@@ -99,6 +113,77 @@ public Film delete(Long filmId) {
+                .collect(Collectors.toList());
+        switch (sortBy) {
+            case "likes":
+                return directorFilms.stream()
```
Сортировку делать лучше на уровне базы

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/storage/FriendStorage.java:
```diff
@@ -42,8 +43,8 @@ public void deleteFriend(Long userId, Long friendId) {
         if (user != null) {
-            String sql = "SELECT u.id, u.email, u.login, u.name, u.birthday FROM users u " +
-                    "JOIN friends f ON f.friend_id = u.id WHERE f.user_id = ?";
+            String sql = "SELECT u.id, u.email, u.login, u.name, u.birthday FROM users u "
```
Можно `select *`

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/storage/LikeStorage.java:
```diff
@@ -39,42 +42,162 @@ public void deleteLike(Long filmId, Long userId) {
+                film.setDescription(rs.getString("description"));
+                film.setReleaseDate(rs.getDate("release_date")
+                        .toLocalDate());
+                film.setDuration(rs.getInt("duration"));
```
Много повторяющегося кода с мапингом, лучше общую часть вынести в метод

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/storage/LikeStorage.java:
```diff
@@ -39,42 +42,162 @@ public void deleteLike(Long filmId, Long userId) {
-                "FROM films LEFT JOIN film_likes ON films.id = film_likes.film_id " +
-                "GROUP BY films.id ORDER BY COUNT(film_likes.user_id) DESC LIMIT ?";
+    public List<Film> getPopular(int count, int genreId, int year) {
+        System.out.println(genreId);
```
Логируем с помощью `log`, если нужно

EvgeniyBushuev committed:
fix: перенести логику добавления события в метод createFeed

krakaduum committed:
refactoring: сортировка фильмов по режиссерам в запросе

DayZscripter committed:
fix: LikeStorage вынесет повторяющийся код в отдельный метод

DayZscripter committed:
fix: codestyle LikeStorage class

Rat-In-Hat committed:
ref,feat

добавить enum с параметрами запроса для поиска фильма
поменять парамметр запроса для поиска фильма с String на List<FilmSearchParameters>

Rat-In-Hat committed:
ref,feat
добавить enum с параметрами запроса для поиска фильма
поменять парамметр запроса для поиска фильма с String на List<FilmSearchParameters>

Rat-In-Hat committed:
del

удалить неиспользуемую реализацию inmemmory

Rat-In-Hat committed:
ref

добавить Db ко всем классам, относящимся к database storage

Rat-In-Hat committed:
ref

перенести пакет model в database, а классы Db в impl (первый раз перенеслось некорректно)

ref

перенести все Db классы в пакет impl

ref

перенести все классы Db в отдельный пакет

Rat-In-Hat committed:
feat

добавить аннотации для отключеня форматирования sql запросов

Rat-In-Hat committed:
style

добавить аннотации

style

повторно применен код стайл, макс длина строки 100 символов

style

применен code style

Rat-In-Hat committed:
Merge remote-tracking branch 'origin/develop' into develop

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java
#	src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/database/impl/FilmDbStorage.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/database/model/FilmSearchParameters.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/inMemory/InMemoryFilmStorage.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/interfaces/FilmStorage.java

alex-from-90 committed:
ТЗ 12.09 add new README.md

alex-from-90 committed:
add new readme

alex-from-90 committed:
ТЗ 12 add new README.md

alex-from-90 committed:
ТЗ 12 add new README.md

alex-from-90 committed:
ТЗ 12 add new README.md

alex-from-90 commented:
Merge full version of develop

bismuthD commented on src/main/java/ru/yandex/practicum/filmorate/model/Feed.java:
```diff
@@ -0,0 +1,21 @@
+
+@Data
+public class Feed {
+    private Long eventId;
```
Вместо множественных private лучше использовать @FieldDefaults

bismuthD commented on src/main/java/ru/yandex/practicum/filmorate/storage/database/impl/FilmDbStorage.java:
```diff
@@ -0,0 +1,189 @@
+
+@Primary // Наставник разрешил использовать вместо @Qualifier
+@Component
+@Slf4j
```
Стоит аннотация для логирования, но нет логирования. Добавьте тогда логирование или уберите аннотацию

bismuthD reviewed (commented):
Отличная работа, очень качественный ход, хорошо разобрались со взаимодействием между слоями сервис контроллер репозиторий, везде использованы верные аннотации, модификатора доступа, оставил пару комментариев на доработку
